lib/app/explore/explore_page.dart

```diff
@@ -120,7 +120,7 @@ class _ExplorePageState extends State<ExplorePage> {
       appBar: YaruWindowTitleBar(
         leading: const SizedBox(width: kLeadingGap),
         title: SearchField(
-          key: ValueKey(showSearchPage),
+          key: ObjectKey(ModalRoute.of(context)?.isCurrent ?? searchQuery),
```

Comment:
Very nice, this works well without causing any rebuild while the page is shown and the query changes

Reply:
oh no... but the other thing that previously worked does not work anymore now :laughing: 
If you click a snap from the explore page without previously searching. The search still works, but now THIS searchquery stays empty :face_exhaling:

Reply:
no this works. But the searchquery does not go away anymore if you clicked on the sidebar to navigate back (also known as "the designer shortcut")

Reply:
If only we had widget tests.. :laughing:

Reply:
Not sure if there's a better way to logically combine `showSearchPage` with the rest:
```suggestion
          key: ValueKey(
            '$showSearchPage${ModalRoute.of(context)?.isCurrent ?? searchQuery}',
          ),
```

Reply:
Thanks, will test!

(Eventually we really need to come back to the tree restructuring. I think the instant search needs to go then want we need an autocomplete search field. )